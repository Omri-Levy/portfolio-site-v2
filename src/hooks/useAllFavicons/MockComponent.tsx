import React from 'react';
import useAllFavicons from '~/hooks/useAllFavicons/useAllFavicons';

const MockComponent: React.FunctionComponent = () => {
	return <div>{JSON.stringify(useAllFavicons())}</div>;
};

export default MockComponent;
