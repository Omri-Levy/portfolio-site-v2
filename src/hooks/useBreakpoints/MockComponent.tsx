import React from 'react';
import useBreakpoints from '.';

const MockComponent: React.FunctionComponent = () => {
	const mediaQuery = useBreakpoints(`md`, `up`);

	return <div>{JSON.stringify(mediaQuery)}</div>;
};

export default MockComponent;
