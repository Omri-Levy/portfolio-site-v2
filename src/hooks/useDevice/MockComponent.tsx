import React from 'react';
import useDevice from '.';

const MockComponent: React.FunctionComponent = () => {
	const { isDesktop } = useDevice();

	return <div>{JSON.stringify(isDesktop)}</div>;
};

export default MockComponent;
