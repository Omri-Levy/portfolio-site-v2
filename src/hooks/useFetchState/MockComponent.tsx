import React from 'react';
import useFetchState from './useFetchState';

const MockComponent: React.FunctionComponent = () => {
	const {
		fetchState,
		setFetchState,
		isLoading,
		shouldDisplayAlert,
		severity,
		alertMessage,
	} = useFetchState({});

	return (
		<>
			{JSON.stringify(fetchState)}
			{JSON.stringify(setFetchState)}
			{JSON.stringify(isLoading)}
			{JSON.stringify(shouldDisplayAlert)}
			{JSON.stringify(severity)}
			{JSON.stringify(alertMessage)}
		</>
	);
};

export default MockComponent;
