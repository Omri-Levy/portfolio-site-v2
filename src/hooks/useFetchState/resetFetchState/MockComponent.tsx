import { Alert } from '@material-ui/lab';
import React, { useState } from 'react';
import resetFetchState from './resetFetchState';

const MockComponent: React.FunctionComponent = () => {
	const [fetchState, setFetchState] = useState<string | undefined>(`success`);

	if (fetchState === undefined) {
		return null;
	}

	return (
		<Alert onClose={resetFetchState(setFetchState, fetchState)}>
			test alert
		</Alert>
	);
};

export default MockComponent;
