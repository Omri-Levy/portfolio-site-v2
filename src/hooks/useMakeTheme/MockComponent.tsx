import React from 'react';
import useMakeTheme from '~/hooks/useMakeTheme/useMakeTheme';

const MockComponent: React.FunctionComponent = () => {
	const theme = useMakeTheme();

	return <div>{JSON.stringify(theme)}</div>;
};

export default MockComponent;
