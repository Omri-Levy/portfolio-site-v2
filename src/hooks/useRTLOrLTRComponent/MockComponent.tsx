import React, { useEffect, useState } from 'react';
import useRTLOrLTRComponent from '.';
import { Props } from './types';
import useThemeContext from '../../context/ThemeProvider/useThemeContext';

const MockComponent: React.FunctionComponent<Props> = ({ rtl }) => {
	const { setIsRTL } = useThemeContext();
	const [isMounted, setIsMounted] = useState(true);

	useEffect(() => {
		if (isMounted) {
			setIsRTL(rtl);
			if (typeof window !== `undefined`) {
				localStorage.setItem(`isRTL`, JSON.stringify(rtl));
			}
		}

		return () => setIsMounted(false);
	}, []);

	return <div>{useRTLOrLTRComponent(<h1>rtl</h1>, <h1>ltr</h1>)}</div>;
};

export default MockComponent;
