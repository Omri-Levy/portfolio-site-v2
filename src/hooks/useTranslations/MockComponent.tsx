import React, { useEffect, useState } from 'react';
import useTranslations from '.';
import { Props } from './types';
import useThemeContext from '../../context/ThemeProvider/useThemeContext';

const MockComponent: React.FunctionComponent<Props> = ({ rtl }) => {
	const [isMounted, setIsMounted] = useState(true);
	const { setIsRTL } = useThemeContext();

	useEffect(() => {
		if (isMounted) {
			setIsRTL(rtl);
			if (typeof window !== `undefined`) {
				localStorage.setItem(`isRTL`, JSON.stringify(rtl));
			}
		}

		return () => setIsMounted(false);
	}, []);

	return <div>{useTranslations(`עברית`, `english`)}</div>;
};

export default MockComponent;
