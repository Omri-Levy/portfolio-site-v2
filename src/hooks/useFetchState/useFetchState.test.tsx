import React from 'react';
import { renderWithProviders } from '../../utils/testUtils';
import MockComponent from './MockComponent';

describe(`useFetchState`, () => {
	it(`matches snapshot`, () => {
		const { container } = renderWithProviders(<MockComponent />);

		expect(container).toMatchSnapshot();
	});
});
