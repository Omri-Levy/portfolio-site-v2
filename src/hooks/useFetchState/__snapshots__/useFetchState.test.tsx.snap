// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useFetchState matches snapshot 1`] = `
<div>
  false
  false
  "error"
  null
</div>
`;
